### PR TITLE
feat: SQL Vaidation + Big Query Cost estimates

### PR DESCRIPTION
**Pull request by suryaiyer95**
## Overview

- Adds basic sql validaiton checks
- Supports Big query cost estimate feature for models

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
\ No newline at end of file
+  "files.exclude": {
+    "out": false // set this to true to hide the "out" folder with the compiled JS files
+  },
+  "search.exclude": {
+    "out": true // set this to false to include "out" folder in search results
+  },
+  // Turn off tsc task auto detection since we have the necessary tasks as npm scripts
+  "typescript.tsc.autoDetect": "off"
+}
```
Comment (mdesmet) [inline .vscode/settings.json:11]: Revert change

File: src/commands/index.ts [hunk 2]
```diff
@@ -16,6 +17,44 @@ import { AltimateScan } from "./altimateScan";
...
+    message += `|${item[0].padEnd(keyLength)}|${item[1].padEnd(
+      valueLength,
+    )}|\r\n`;
+  }
+  message +=
+    "|" + "_".repeat(keyLength) + "|" + "_".repeat(valueLength) + "|" + "\r\n";
+  message += "\r\n";
+
+  return message;
+};
```
Comment (mdesmet) [inline src/commands/index.ts:57]: If only used in VscodeCommands, make it a private method.

File: src/commands/index.ts [hunk 3]
```diff
@@ -48,6 +89,16 @@ export class VSCodeCommands implements Disposable {
       commands.registerCommand("dbtPowerUser.compileCurrentModel", () =>
         this.runModel.compileModelOnActiveWindow(),
       ),
+      commands.registerCommand("dbtPowerUser.costEstimate", async () => {
```
Comment (mdesmet) [inline src/commands/index.ts:92]: Maybe add bigquery in the name as it is really only relevant to bigquery

File: src/manifest/dbtProject.ts [hunk 4]
```diff
@@ -516,6 +517,96 @@ export class DBTProject implements Disposable {
     return output.compiled_sql;
   }
 
+  public async validateSql(request: {
```
Comment (mdesmet) [inline src/manifest/dbtProject.ts:520]:
```suggestion
 async validateSql(request: {
```

Please remove public modifiers.

File: src/manifest/dbtProject.ts [hunk 5]
```diff
@@ -516,6 +517,96 @@ export class DBTProject implements Disposable {
...
+    }
+    try {
+      const { sql, dialect, models } = request;
+      const result = await this.python?.lock(
+        (python) =>
+          python!`to_dict(validate_sql(${sql}, ${dialect}, ${models}))`,
+      );
+      return result as ValidateSqlParseErrorResponse;
+    } catch (exc) {
+      console.log(exc);
```
Comment (mdesmet) [inline src/manifest/dbtProject.ts:546]: I think this needs exception handling, telemetry and showErrorMessage.

File: src/manifest/dbtProject.ts [hunk 6]
```diff
@@ -516,6 +517,96 @@ export class DBTProject implements Disposable {
...
+    }
+    try {
+      const result = await this.python?.lock(
+        (python) => python!`to_dict(project.get_dbt_version())`,
+      );
+      return (result as number[]) || [0, 0, 0];
+    } catch (exc) {
+      console.log(exc);
+    }
+    return [0, 0, 0];
```
Comment (mdesmet) [inline src/manifest/dbtProject.ts:607]: Why return 0,0,0. I think the exception should be handled in the calling code.

File: src/quickpick/sqlQuickPick.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        );
+        dbtpuquickpick.show();
+      });
+    } finally {
+      disposables.forEach((d) => d.dispose());
+    }
+  }
+}
+
+class DbtPowerUserControlPanelItem implements QuickPickItem {
```
Comment (mdesmet) [inline src/quickpick/sqlQuickPick.ts:93]: Rename accordingly, this is not the DbtPowerUserControlPanel

File: src/commands/index.ts [hunk 8]
```diff
@@ -16,6 +17,44 @@ import { AltimateScan } from "./altimateScan";
...
+  const n = modelName.length;
+  if (maxKeyLength + maxValueLength + 1 < n) {
+    keyLength = Math.floor(
+      (n * maxKeyLength) / (maxKeyLength + maxValueLength),
+    );
+    valueLength = n - keyLength;
+  }
+  let message = "";
+  message += " " + "_".repeat(keyLength + valueLength + 1) + " " + "\r\n";
+  message += "|" + modelName.padEnd(keyLength + valueLength + 1) + "|" + "\r\n";
```
[the remark this answers is not in the document]
Comment (AdiGajbhiye) [reply src/commands/index.ts]: I have added model name as title. Since the new message is appended.

**Review by mdesmet — APPROVED** [review]

**Pull request merged**